pkg/resource/deploy/source_query.go

```diff
@@ -168,15 +198,61 @@ func runLangPlugin(src *querySource) result.Result {
 
 // newQueryResourceMonitor creates a new resource monitor RPC server intended to be used in Pulumi's
 // "query mode".
-func newQueryResourceMonitor(builtins *builtinProvider, tracingSpan opentracing.Span) (*queryResmon, error) {
+func newQueryResourceMonitor(
+	builtins *builtinProvider, defaultProviderVersions map[tokens.Package]*semver.Version,
+	provs ProviderSource, reg *providers.Registry, plugctx *plugin.Context,
+	providerRegErrChan chan<- result.Result, tracingSpan opentracing.Span) (*queryResmon, error) {
 
 	// Create our cancellation channel.
 	cancel := make(chan bool)
 
+	// Create channel for handling registrations.
+	regChan := make(chan *registerResourceEvent)
+
+	// Create a new default provider manager.
+	var config *Target
+	d := &defaultProviders{
+		defaultVersions: defaultProviderVersions,
+		providers:       make(map[string]providers.Reference),
+		config:          config,
+		requests:        make(chan defaultProviderRequest),
+		regChan:         regChan,
+		cancel:          cancel,
+	}
+
+	go func() {
+		for {
+			select {
+			case e := <-regChan:
+				urn := resource.NewURN(
+					"query-stack", "query-project", "parent-type", e.goal.Type, e.goal.Name)
+
+				inputs, _, err := reg.Check(urn, resource.PropertyMap{}, e.goal.Properties, false)
```

Comment:
We should probably include `CheckFailures` in the result here, but I'm still working through how to flow that back through the results.